### DISCUSSION TITLE
sol.exe is marked as trojan script by Windows Security

### DISCUSSION THREAD

**Discussion by Hruda-Rockey10** [cpeditor/cpeditor#1383] [Q&A] [answered]
<img width="592" alt="tjp" src="https://github.com/user-attachments/assets/4be0dbe1-e78f-4c7d-9484-9d4dc103ea9c">

windows security shows its  a trojan script and it creates issue like shows TLE and RE and codes doesn't execute during my coding time

**Comment by ouuan** [comment]
Does this happen for all codes or some specific problem? Can you post your code here?

**Comment by Hruda-Rockey10** [comment]
it happens for all code
right now i dont have ss ..if i get error i will post the ss here

**Comment by Hruda-Rockey10** [comment]
<img width="1118" alt="git1" src="https://github.com/user-attachments/assets/d0d5d6af-8d1f-4b6a-b2f1-ae559da35aa2">
@ouuan 
i just did it now 

it worked showing AC 
but many times it shoes RE TLE

**Comment by ouuan** [comment]
1. Compile it again.
2. Find the executable file. If you don't know where it is, right-click on the tab to open the menu, and then choose "reveal executable file".
   ![image](https://github.com/user-attachments/assets/74e387d1-02f6-4f4f-9535-93d61dc1fa9e)
3. Upload the executable file to https://www.virustotal.com/ and see the results.

If Windows Security pops up a notification during these steps, try to click on the notification and stop it from deleting the executable file.

**Reply by Hruda-Rockey10** [reply]
okay let me check

**Reply by Hruda-Rockey10** [reply]
1. when i got that error ( i clicked on reveal Executable File) which showed two options
<img width="674" alt="git 2" src="https://github.com/user-attachments/assets/4efafe37-97fd-4225-a8bc-f54d4641203b">

2. out of which i uploaded the sol(cp) file to virustotal  and got a green sign (nothing malicious) but i am not sure about the sol(application) file because there is no option for application file in virus total.

3.  In windows security it automatically quarantines the file and shows   3 options ( restore , remove, allow) 
 @ouuan

**Reply by ouuan** [reply]
> i am not sure about the sol(application) file because there is no option for application file in virus total.

I don't understand. What is "no option"? Just upload it. Do you mean there is a file type filter in the upload dialog? Please post a screenshot.

**Reply by Hruda-Rockey10** [reply]
i uploaded the sol( application) file in virus total  this are the results 
<img width="1120" alt="git 3" src="https://github.com/user-attachments/assets/89ea3b40-9229-4e42-8dd1-a7a0a9e53f24">
@ouuan

**Comment by ouuan** [comment]
Please post the complete C++ code.

**Reply by Hruda-Rockey10** [reply]
```cpp
// Problem: B. XOR Sequences
// Contest: Codeforces - Codeforces Round 951 (Div. 2)
// URL: https://codeforces.com/contest/1979/problem/B
// Memory Limit: 256 MB
// Time Limit: 1000 ms

#include <bits/stdc++.h>
using namespace std;
#define int long long
#define pii pair<int,int>
#define fi first
#define se second
#define pb push_back
#define mp make_pair  //v.pb(mp(x,y));--> vector of pair
#define range(i,j,m,n) (i>=0 && i<m && j>=0 && j<n)
#define f(i, n) for (int i = 0; i < n; i++)
#define f1(i, n) for (int i = 1; i < n+1; i++)
#define max_ele(v) *max_element(v.begin(), v.end())
#define min_ele(v) *min_element(v.begin(), v.end())
#define fsort(v) sort(v.begin(), v.end())
#define rsort(v) sort(v.begin(), v.end(), greater<int>())
const int  mod = 1e9+7;
//To insert values in pair 
// then you write make_pair(i,j) or v.pb({i,j}) or v[i]=make_pair(i,j)
// Acess values you write v[i].first and v[i].second.

void hruda ()
{
  	
	int x,y;
  	cin>>x>>y;
  	//int cnt=0;int ans=0;
  	for(int i=0;i<32;i++)
  	{	
  		if((x&(1<<i)) != (y&(1<<i)))
  		{
  			cout<<(1<<i)<<endl;
  			//cout<<i<<" "<<x<<" "<<y<<endl;
  			return;
  		}
  	}
  	return;
	
}

int32_t main()
{   
	ios_base::sync_with_stdio(NULL);
	cin.tie(NULL);
	cout.tie(NULL);
	
	int t;
	cin>>t;
	while(t--)
	{
		hruda();
	}
	return 0;
}
```

**Comment by ouuan** [comment]
Please try changing the code to a plain `int main(){}` and try again. Is it still detected as malware?

**Reply by Hruda-Rockey10** [reply]
1.It  is working fine for now with `int main(){}` and works smoothly
2. But it was working fine since 1 year with int32_t but its fine i have to change the whole template
3.  Btw i am not sure its fixed  ..give me time till midnight i will let u know if its works fine completely with `int main()`

**Reply by ouuan** [reply]
`int` and `int32_t` should be the same. My point is the empty function body `{}` and I did not even notice that it was `int32_t`. Do you mean that the same function body is detected as malware when using `int32_t` but not when using `int`?

**Reply by Hruda-Rockey10** [reply]
the issue is happening again with int 
<img width="1119" alt="git 4" src="https://github.com/user-attachments/assets/116b3174-abd6-4b68-ab41-59719ad8bc6d">

**Reply by Hruda-Rockey10** [reply]
`// Contest: Codeforces - Codeforces Round 951 (Div. 2)
// URL: https://codeforces.com/contest/1979/problem/B
// Memory Limit: 256 MB
// Time Limit: 1000 ms

#include <bits/stdc++.h>
using namespace std;
//#define int long long
#define pii pair<int,int>
#define fi first
#define se second
#define pb push_back
#define mp make_pair  //v.pb(mp(x,y));--> vector of pair
#define range(i,j,m,n) (i>=0 && i<m && j>=0 && j<n)
#define f(i, n) for (int i = 0; i < n; i++)
#define f1(i, n) for (int i = 1; i < n+1; i++)
#define max_ele(v) *max_element(v.begin(), v.end())
#define min_ele(v) *min_element(v.begin(), v.end())
#define fsort(v) sort(v.begin(), v.end())
#define rsort(v) sort(v.begin(), v.end(), greater<int>())
const int  mod = 1e9+7;
//To insert values in pair 
// then you write make_pair(i,j) or v.pb({i,j}) or v[i]=make_pair(i,j)
// Acess values you write v[i].first and v[i].second.

void hruda ()
{
  	
	int x,y;
  	cin>>x>>y;
  	//int cnt=0;int ans=0;
  	for(int i=0;i<32;i++)
  	{	
  		if((x&(1<<i)) != (y&(1<<i)))
  		{
  			cout<<(1<<i)<<endl;
  			//cout<<i<<" "<<x<<" "<<y<<endl;
  			return;
  		}
  	}
  	return;
	
}

int main()
{   
	ios_base::sync_with_stdio(NULL);
	cin.tie(NULL);
	cout.tie(NULL);
	
	int t;
	cin>>t;
	while(t--)
	{
		hruda();
	}
	return 0;
}`

**Reply by ouuan** [reply]
Please, I have already explained that the point is the empty function body, and `int`/`int32_t` does not matter. So please try `int main(){}` as the whole code, with no function body.

**Reply by Hruda-Rockey10** [reply]
okay let me try ..

**Reply by Hruda-Rockey10** [reply]
<img width="1106" alt="git 5" src="https://github.com/user-attachments/assets/27208389-e5a5-473d-b7a7-71292c83d105">

getting warning from windows security same

**Comment by ouuan** [comment]
What compiler are you using? How did you install it? Can you try compiling the code without using CP Editor?

**Reply by Hruda-Rockey10** [reply]
i dont know properly becz i installed the compiler and eveything many years back

**Comment by Hruda-Rockey10** [comment, marked as answer]
@ouuan  i uninstalled the mingw (old version) and installed the cp editor with GCC LLVM  and its working fine without any error. 
Thansk for ur time and help

**Comment by ouuan** [comment]
Do you experience slow execution time for the first run after each compilation?

**Reply by Hruda-Rockey10** [reply]
yess i experience slow execution time .. is there any solution?

**Reply by ouuan** [reply]
Try https://cpeditor.org/docs/faq/#my-code-gets-tle-on-every-first-execution-after-compilation

**Comment by Hruda-Rockey10** [comment]
@ouuan 
i was getting the same issue again even after downloading new mingw ,  after googling the issue i got this information from Microsoft discussion .  ( Uploading the pic so that others can find solution if they have the issue ).
<img width="759" alt="trj5" src="https://github.com/user-attachments/assets/121db590-3564-4d18-a3d2-319993338331">
